test/functional/test_runner.py

```diff
@@ -702,6 +716,17 @@ def print_results(test_results, max_len_name, runtime):
     results += "Runtime: %s s\n" % (runtime)
     print(results)
 
+
+def write_results(test_results, filepath, total_runtime):
+    with open(filepath, mode="w", encoding="utf8") as results_file:
+        results_writer = csv.writer(results_file)
+        results_writer.writerow(['test', 'status', 'duration(seconds)'])
+        all_passed = True
+        for test_result in test_results:
+            all_passed = all_passed and test_result.was_successful
+            results_writer.writerow([test_result.name, test_result.status, str(test_result.time)])
+        results_writer.writerow(['ALL', ("Passed" if all_passed else "Failed"), str(total_runtime)])
```

Comment:
Nit: Having had only few tests failed also display `ALL,Failed`, which in the first glance gives the impression that all of them failed. In cases like this, displaying `ALL, Few failed` would be more explicit.

Reply:
Yeah, at first glance showing `ALL,Failed` is not ideal, but in this case the CSV output is matching the approach taken by stdout printing, so it seems appropriate for now.

Since the scope of this PR is to focus on adding the CSV output capability, I'll leave it as-is for now.  Perhaps a future discussion for how `ALL` is treated in both the stdout printing and the CSV could be had?

```
TEST                   | STATUS    | DURATION

feature_blocksdir.py   | ✓ Passed  | 1 s
feature_config_args.py | ✓ Passed  | 29 s
wallet_startup.py      | ✓ Passed  | 3 s
mempool_accept.py      | ✖ Failed  | 1 s

ALL                    | ✖ Failed  | 34 s (accumulated) 
Runtime: 29 s
```

Reply:
I see, makes sense to have that discussion separate from this.